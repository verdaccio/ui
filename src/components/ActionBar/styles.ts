import styled from 'react-emotion';
import { default as MuiFab } from '@material-ui/core/Fab';

import colors from '../../utils/styles/colors';
import ListItem from '../../muiComponents/ListItem';

export const ActionListItem = styled(ListItem)({
  paddingTop: 0,
  paddingLeft: 0,
  paddingRight: 0,
});

export const Fab = styled(MuiFab)({
  '&&': {
    backgroundColor: colors.primary,
    color: colors.white,
    marginRight: '10px',
  },
});
