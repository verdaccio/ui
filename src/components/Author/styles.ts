import styled from '@emotion/styled';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import ListItem from '../ListItem';
import ListItemText from '../ListItemText';
import Text from '../Text';

export const StyledText = styled(Text)<{ theme?: Theme }>(props => ({
  fontWeight: props.theme && props.theme.fontWeight.bold,
}));

export const AuthorListItem = styled(ListItem)({
  padding: 0,
  ':hover': {
    backgroundColor: 'transparent',
  },
});

export const AuthorListItemText = styled(ListItemText)({
  padding: '0 10px',
  margin: 0,
});
