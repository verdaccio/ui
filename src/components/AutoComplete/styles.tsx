import styled from '@emotion/styled';
import React from 'react';

import { Theme } from 'verdaccio-ui/design-tokens/theme';

import Paper from '../Paper';
import TextField from '../TextField';

export interface InputFieldProps {
  color: string;
}

export const Wrapper = styled('div')({
  width: '100%',
  height: '32px',
  position: 'relative',
  zIndex: 1,
});

export const StyledTextField = styled(TextField)<{ theme?: Theme }>(props => ({
  '& .MuiInputBase-root': {
    ':before': {
      content: "''",
      border: 'none',
    },
    ':after': {
      borderColor: props.theme && props.theme.palette.white,
    },
    ':hover:before': {
      content: 'none',
    },
  },
  '& .MuiInputBase-input': {
    color: props.theme && props.theme.palette.white,
  },
}));

/* eslint-disable verdaccio/jsx-spread */
// @ts-ignore types of color are incompatible
export const InputField: React.FC<InputFieldProps> = ({ ...others }) => <StyledTextField {...others} />;

export const SuggestionContainer = styled(Paper)({
  maxHeight: '500px',
  overflowY: 'auto',
});
