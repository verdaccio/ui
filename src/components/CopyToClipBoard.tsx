import styled from '@emotion/styled';
import FileCopy from '@material-ui/icons/FileCopy';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { copyToClipBoardUtility } from 'verdaccio-ui/utils/cli-utils';

import IconButton from './IconButton';
import Tooltip from './Tooltip';

interface Props {
  text: string;
  children?: React.ReactNode;
}

function CopyToClipBoard({ text, children, ...props }: Props) {
  const { t } = useTranslation();
  return (
    <Wrapper {...props}>
      <Content>{children ?? text}</Content>
      <Tooltip disableFocusListener={true} title={t('copy-to-clipboard')}>
        <IconButton onClick={copyToClipBoardUtility(text)} data-testid="copy-icon">
          <FileCopy />
        </IconButton>
      </Tooltip>
    </Wrapper>
  );
}

export default CopyToClipBoard;

const Wrapper = styled('div')({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
});

const Content = styled('span')({
  display: 'inline-block',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
  textOverflow: 'ellipsis',
  height: '21px',
  fontSize: '1rem',
});
