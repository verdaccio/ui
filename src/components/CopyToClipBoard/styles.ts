import styled from '@emotion/styled';

import IconButton from '../IconButton';

export const ClipBoardCopy = styled('div')({
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
});

export const ClipBoardCopyText = styled('span')({
  display: 'inline-block',
  textOverflow: 'ellipsis',
  overflow: 'hidden',
  whiteSpace: 'nowrap',
  height: '21px',
  fontSize: '1rem',
});

export const CopyIcon = styled(IconButton)({});
