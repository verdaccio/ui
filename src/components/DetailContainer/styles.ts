import styled from 'react-emotion';

export const Content = styled('div')`
  && {
    padding: 15px;
  }
`;
