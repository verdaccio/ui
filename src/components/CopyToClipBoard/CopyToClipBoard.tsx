import FileCopy from '@material-ui/icons/FileCopy';
import React from 'react';
import { useTranslation } from 'react-i18next';

import { copyToClipBoardUtility } from 'verdaccio-ui/utils/cli-utils';

import Tooltip from '../Tooltip';

import { ClipBoardCopy, ClipBoardCopyText, CopyIcon } from './styles';

interface Props {
  text: string;
  children?: React.ReactNode;
}

const renderText = (text: string, children: React.ReactNode): JSX.Element => {
  if (children) {
    return <ClipBoardCopyText>{children}</ClipBoardCopyText>;
  }

  return <ClipBoardCopyText>{text}</ClipBoardCopyText>;
};

const CopyToClipBoard: React.FC<Props> = ({ text, children }) => {
  const { t } = useTranslation();
  return (
    <ClipBoardCopy>
      {renderText(text, children)}
      <Tooltip disableFocusListener={true} title={t('copy-to-clipboard')}>
        <CopyIcon onClick={copyToClipBoardUtility(text)}>
          <FileCopy />
        </CopyIcon>
      </Tooltip>
    </ClipBoardCopy>
  );
};

export default CopyToClipBoard;
