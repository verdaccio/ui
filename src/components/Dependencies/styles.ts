import styled from 'react-emotion';

import { fontWeight } from '../../utils/styles/sizes';
import Text from '../../muiComponents/Text';
import Card from '../../muiComponents/Card';
import Chip from '../../muiComponents/Chip';

export const CardWrap = styled(Card)({
  margin: '0 0 16px',
});

export const StyledText = styled(Text)({
  fontWeight: fontWeight.bold,
  textTransform: 'capitalize',
});

export const Tags = styled('div')({
  display: 'flex',
  justifyContent: 'start',
  flexWrap: 'wrap',
  margin: '0 -5px',
});

export const Tag = styled(Chip)({
  margin: '5px',
});
