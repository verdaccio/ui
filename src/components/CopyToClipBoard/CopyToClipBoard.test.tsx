import { ReactWrapper } from 'enzyme';
import React from 'react';

import { copyToClipBoardUtility } from 'verdaccio-ui/utils/cli-utils';
import { mount } from 'verdaccio-ui/utils/test-enzyme';

import CopyToClipBoard from './CopyToClipBoard';
import { CopyIcon } from './styles';

jest.mock('verdaccio-ui/utils/cli-utils');

describe('<CopyToClipBoard /> component', () => {
  let wrapper: ReactWrapper;
  const copyText = 'copy text';

  beforeEach(() => {
    wrapper = mount(<CopyToClipBoard text={copyText} />);
  });

  test('should load the component in default state', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call the copyToClipBoardUtility for copy to clipboard utility', () => {
    wrapper.find(CopyIcon).simulate('click');
    expect(copyToClipBoardUtility).toHaveBeenCalledWith(copyText);
  });
});
