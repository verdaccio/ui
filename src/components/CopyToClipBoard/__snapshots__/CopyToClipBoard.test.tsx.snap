// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CopyToClipBoard /> component should load the component in default state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-0 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  height: 21px;
  font-size: 1rem;
}

<body>
    <div>
      <div
        class="emotion-4 emotion-5"
      >
        <span
          class="emotion-0 emotion-1"
        >
          copy text
        </span>
        <button
          class="MuiButtonBase-root MuiIconButton-root  emotion-2 emotion-3"
          data-testid="copy-icon"
          tabindex="0"
          title="Copy to clipboard"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": .emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-0 {
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  height: 21px;
  font-size: 1rem;
}

<div>
    <div
      class="emotion-4 emotion-5"
    >
      <span
        class="emotion-0 emotion-1"
      >
        copy text
      </span>
      <button
        class="MuiButtonBase-root MuiIconButton-root  emotion-2 emotion-3"
        data-testid="copy-icon"
        tabindex="0"
        title="Copy to clipboard"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
